### DISCUSSION TITLE
Unable to add a Vector Store to a new Assistant thread.

### DISCUSSION THREAD

**Discussion by joycpa** [openai/openai-dotnet#279] [General]
Am trying to add a vector store to a new Assistant thread. Have added the vectorStore.Id to the fileSearchToolResources and then included it in the ThreadCreationOptions. Following is the code. Am getting a null reference error. 

//Add vector store to thread
FileSearchToolResources fileSearchToolResources = new FileSearchToolResources();
fileSearchToolResources.NewVectorStores.Clear();
fileSearchToolResources.VectorStoreIds.Add(vectorStore.Id);

ThreadRun threadRun;
try
{

    ThreadCreationOptions threadOptions = new()
    { 
        ToolResources =
        {
            FileSearch = fileSearchToolResources,
        },

        InitialMessages = { "New file uploaded " + FileUploaded.Id + " Create Profit and Loss Statement from this uploaded file. " }

    };

    threadRun = assistantClient.CreateThreadAndRun(_globalSettings.AssistantId, threadOptions);

}

**Comment by cpalm1974** [comment]
Has anyone solved this?
I've got the same issue with AssistantCreationOptions

`  AssistantCreationOptions options = assistant.GetAssistantCreationOptions();
  options.ToolResources.FileSearch.VectorStoreIds.Add(vectorStore?.vectorStore.Id);`

I see that the problem relies in VectorStoreIds being an unitialized list. 
Due to the fact that the set method is declared internal it's not possible to assign an empty list to it.
So I assume this might be a bug as either the initialization in the constructor is missing (so that you can work with the methods or declaration as interal is wrong here:

FileSearchToolResources.cs
`    public IList<string> VectorStoreIds
    {
        get => _vectorStoreIds;
        internal set
        {
            _vectorStoreIds = new ChangeTrackingList<string>();
            foreach (string item in value)
            {
                _vectorStoreIds.Add(item);
            }
        }
    }`

Does anyone know a workaround?
